unshare/unshare.go

```diff
@@ -40,6 +40,7 @@ type Cmd struct {
 // Command creates a new Cmd which can be customized.
 func Command(args ...string) *Cmd {
 	cmd := reexec.Command(args...)
+	cmd.Env = append(os.Environ(), "__rootless__=1")
```

Comment:
Setting this field deviates from the behavior of the function from the `os/exec` package that we're mimicking.  This would also be set before the caller could customize the object, including for cases where `UnshareFlags` isn't going to have `CLONE_NEWUSER` set.  It would break the current caller's current logic for setting `_BUILDAH_STARTED_IN_USERNS` in the child's environment, which it does by setting the value it in its own environment after creating the `Command` object.  Is there a reason we're not keying off of that variable?

Reply:
Nope, I wish you told me about this a while ago.